### ISSUE TITLE
How to do knowledge distillation training

### ISSUE THREAD

**Issue by RyanOvO** [huggingface/open-r1#240]
In the deepseek r1 technical report, there is a small model based on distillation at the end; deepseek r1, as the teacher model,  qwen and llama, as the student model, do SFT based on distilled data. However, it seems that the process of knowledge distillation is not involved here(open r1), that is, the process of the r1 teacher model modifying the output of the student model, but simply SFT based on distilled data.

**Comment by caijianwei1996** [comment]
SFT stage dataset:”HuggingFaceH4/Bespoke-Stratos-17k“,comes from deepseek r1.You can also generate new dataset by generate.py

**Comment by QuantaoYao** [comment]
I have some confusion regarding the distillation implementation details of deepseekr1. The reference answers are generated by a teacher model (such as deepseekr1-zero) and knowledge is transferred through softened labels (such as probability distributions). However, when looking at the specific reproduction, it seems to adopt a one-question-multiple-answers format. For a given question, deepseekr1-zero outputs 4 answers, and then these four answers with <think></think> tags are used as the answer for traditional SFT. This does not seem to reflect the process of label softening. How is this specifically implemented?

**Comment by cs-whh** [comment]
In their report : "we directly fine-tuned open-source models like Qwen (Qwen, 2024b) and Llama (AI@Meta, 2024) using
the 800k samples curated with DeepSeek-R1". So I think the distillation is the standary SFT.

**Comment by hackape** [comment]
I have the same confusion. Doesn’t the ML field have a consensus on the definition of “distillation”? The key aspect of distillation is learning from the softened labels in the teacher model’s output rather than the hard labels in the training dataset. Without that element, it shouldn’t be considered distillation. Why not just call it “SFT with synthetic data”? That would avoid a lot of confusion.

**Comment by ggcr** [comment]
IMO if the student is not trying to approximate the underlying distribution of the teacher via the **logits** then it has nothing to do with Knowledge Distillation as per Hinton [1]

In any case, it might be "Distillation", without the "Knowledge" part.

[1] Hinton, Geoffrey, Oriol Vinyals, and Jeff Dean. "Distilling the knowledge in a neural network." arXiv preprint arXiv:1503.02531 (2015).

---

Llama 3, in their blog also mentioned Distillation, but later on was confirmed that it was not KD.
